src/rgw/rgw_lua_utils.cc

```diff
@@ -24,21 +19,10 @@ int RGWDebugLog(lua_State* L)
 {
   auto cct = reinterpret_cast<CephContext*>(lua_touserdata(L, lua_upvalueindex(1)));
 
-  constexpr auto NUM_RETURN = 0;
-
-  if (!lua_isstring(L, -1)) {
-    if (cct) {
-      ldout(cct, 1) << "Lua ERROR: missing/invalid 'message' parameter when calling Log"  << dendl;
-    }
-    return NUM_RETURN;
-  }
-  const char* message = lua_tostring(L, -1);
-  if (cct) {
-    ldout(cct, 20) << "Lua INFO: " << message << dendl;
-  }
-
-  return NUM_RETURN;
-};
+  auto message = luaL_checkstring(L, 1);
```

Comment:
what if we pass `nil` to the function?
we don't want to call `ldout` if `luaL_checkstring()` return a null pointer

Reply:
can you add to the unit test file 2 tests:
- one that validate that we can handle nil:
```lua
TEST(TestRGWLua, RGWDebugLogError)
{
  const std::string script = R"( 
    RGWDebugLog(nil)
  )"; 

  DEFINE_REQ_STATE;

  const auto rc = lua::request::execute(nullptr, nullptr, nullptr, &s, "", script);
  ASSERT_EQ(rc, -1);
}
```
- one that test we can handle non-strings after your fix:
```lua
TEST(TestRGWLua, RGWDebugLogNumber)
{
  const std::string script = R"( 
    RGWDebugLog(1234567890)
  )"; 

  DEFINE_REQ_STATE;

  const auto rc = lua::request::execute(nullptr, nullptr, nullptr, &s, "", script);
  ASSERT_EQ(rc, 0);
}
```

Reply:
in addition, please update: https://github.com/ceph/ceph/blob/master/doc/radosgw/lua-scripting.rst#debug-log
since the log action is not restricted to string inputs anymore

Reply:
> what if we pass nil to the function?

You'll get an error, like:

````
$ lua
Lua 5.4.0  Copyright (C) 1994-2020 Lua.org, PUC-Rio
> io.write(nil)
stdin:1: bad argument #1 to 'write' (string expected, got nil)
stack traceback:
        [C]: in function 'io.write'
        stdin:1: in main chunk
        [C]: in ?
````

I don't think it's desirable to accept nil here?

Reply:
agree. just wanted to make sure we don't fail the c++ code

Reply:
Added tests but left the `nil` test as expecting an error.